pkg/monitor/cluster/arooperatorheartbeat.go

```diff
@@ -0,0 +1,28 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"strconv"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (mon *Monitor) emitAroOperatorHeartbeat(ctx context.Context) error {
+	aroDeployments, err := mon.cli.AppsV1().Deployments("openshift-azure-operator").List(metav1.ListOptions{})
+
+	if err != nil {
+		return err
+	}
+
+	for _, d := range aroDeployments.Items {
+		mon.emitGauge("arooperator.heartbeat", 1, map[string]string{
+			"name":      d.Name,
+			"available": strconv.FormatBool(d.Status.AvailableReplicas == d.Status.Replicas),
```

Comment:
> d.Status.AvailableReplicas == d.Status.Replicas

Does this mean a deployment will be marked as unavailable, when some replicas are down while others are still healthy? During a rolling update, for example, K8s will still consider a deployment as healthy even though the number of available replicas is not equal to the desired replicas.

Reply:
This is a valid point. While operator deployments only have 1 replica and the updates are very rare, avoiding the blips in deployment availability is probably a good idea. We can use `availableReplicas` or `readyReplicas` instead of `replicas`:
```
$ set image deploy hello-openshift  hello-openshift=openshift/hello-openshift:v3.9
deployment.apps/hello-openshift image updated
$ while :; do oc get deploy hello-openshift -o jsonpath="{.status.replicas}{'\t'}{.status.availableReplicas}{'\t'}{.status.readyReplicas}{'\t'}{.status.updatedReplicas}{'\n'}" && sleep 1; done
2       1       1       1
...
1       1       1       1
```
Definitions: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#deploymentstatus-v1-apps

Reply:
How about `unavailableReplicas == replicas` implying that all replicas are unavailable?

Reply:
I used a confusing dimension name. The goal over here is to have a metric that shows if the aro-operator is completely healthy, so even 1 replica down should turn it to false.

I changed the dimension name to "healthy" (still has true or false as values) and I check if unavailableReplicas!=0. The current implementation covers the upgrade case, because unavailable replicas don't go negative.